### PR TITLE
workflows: use brew pr-pull

### PR DESCRIPTION
**Pull request by dawidd6**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

Short discussion about the idea here: https://github.com/Homebrew/homebrew-core/pull/52371

cc @jonchang

### CODE REVIEW

**Review by jonchang — APPROVED** [review]
Looks good to me!

File: .github/workflows/publish-commit-bottles.yml [hunk 1]
```diff
@@ -23,51 +23,30 @@ jobs:
...
           GIT_COMMITTER_EMAIL: ${{steps.env.outputs.email}}
         run: |
-          cd $(brew --repo ${{github.repository}})
-          git rebase --force origin/master
-          MESSAGE="$(git log HEAD^.. --format=%B)
-
-          Closes #${{github.event.client_payload.pull_request}}."
-          git commit --amend --signoff --allow-empty -m "$MESSAGE"
-          git show --pretty=fuller
+          brew pr-pull ${{github.event.client_payload.pull_request}}
```
Comment (dawidd6) [inline .github/workflows/publish-commit-bottles.yml]: Do we need to set `HOMEBREW_GITHUB_API_TOKEN` for `brew pr-pull`?

Comment (jonchang) [reply .github/workflows/publish-commit-bottles.yml]: I’m not sure if GITHUB_TOKEN is sufficient. It should need the same secrets as download-artifact I believe?

Comment (dawidd6) [reply .github/workflows/publish-commit-bottles.yml]: Guess we can merge it now, test it in practice and fix later if needed.

**Pull request merged**